Sources/Apodini/Components/Component.swift

```diff
@@ -26,8 +26,24 @@ public protocol Component {
     
     /// A function that is called when a request reaches the `Component`
     func handle() -> Self.Response
+    /// The default implementation simply wraps results from
+    /// `handle() -> Self.Response` into `.final(..)`.
+    func handle() -> Action<Self.Response>
```

Comment:
@nerdsupremacist I am having a hard time trying to get rid of the second `handle` function, as we discussed 🤔
If I make `Action: ResponseEncodable`, and only have the `handle() -> Response` function, I still do not see how we could retrieve the real type of the returned Action's generic. Even if I would be able to check the returned `Response` for conformance to `Action` with the code you showed me.
But as far as I see, we need this type-info to be able to unwrap the value from `.final(..)` and `.send(..)`.

After thinking this through again, I think we should discuss whether having specialized types of handlers for different kinds of communicational patterns is an option. In my opinion, it would be a quite useful thing, as it removes the ambiguity of the returned `Response`s type, and it would also hinder the developer to overwrite both of the `handle` functions (as is possible to do in my current version).

Reply:
Hmmmm I see. 
1: I personally think that we need to go a different route from `ResponseEncodable`. 
Because stating that an action is response encodable is just wrong semantically speaking.
2: As for the generic type of action, we can probably work around that with an internal protocol like this:
```
protocol ActionProtocol {
    static var encodedType: Any.Type { get }
    var value: Any? { get }
    var isFinal: Bool { get }
}
```
I think we can keep this protocol internal if we decide to only use to build the semantic model
3: The idea of a different kind of handler also works for me and seems intuitive enough